cmd/cosign/cli: some files aren't closed before they are deleted

**Description**

I ran one of Orijtech Inc's tools, staticmajor on this repo and it reported
```shell
$ staticmajor -resleak=true -resleak.fuzzy=true ./...
/Users/emmanuelodeke/go/src/github.com/sigstore/cosign/cmd/cosign/cli/policy_init.go:144:35: leaking resource
/Users/emmanuelodeke/go/src/github.com/sigstore/cosign/cmd/cosign/cli/policy_init.go:307:35: leaking resource
```
with credible reports
https://github.com/sigstore/cosign/blob/8d2a1a6e16f0375c05314481206cc591c51c041e/cmd/cosign/cli/policy_init.go#L144-L150\

and 

https://github.com/sigstore/cosign/blob/8d2a1a6e16f0375c05314481206cc591c51c041e/cmd/cosign/cli/policy_init.go#L307-L312

for which there are credible leaks that could even be used to prevent a file deletion if one finagled that file handle to the process in the time before the defer os.Remove(tempFile.Name()) was invoked say on Windows

**Version**
At commit 8d2a1a6e16f0375c05314481206cc591c51c041e
<!--What version are you using? e.g. `cosign version`...
-->


**Remedy**
Invoke tempFile.Close() before os.Remove(tempFile.Name()) per
```diff
diff --git a/cmd/cosign/cli/policy_init.go b/cmd/cosign/cli/policy_init.go
index 1f4be09c..b05b6dd7 100644
--- a/cmd/cosign/cli/policy_init.go
+++ b/cmd/cosign/cli/policy_init.go
@@ -146,7 +146,10 @@ func initPolicy() *cobra.Command {
 					return err
 				}
 				outfile = tempFile.Name()
-				defer os.Remove(tempFile.Name())
+				defer func() {
+					tempFile.Close()
+					os.Remove(tempFile.Name())
+				}()
 			}
 
 			files := []cremote.File{
@@ -309,7 +312,10 @@ func signPolicy() *cobra.Command {
 					return err
 				}
 				outfile = tempFile.Name()
-				defer os.Remove(tempFile.Name())
+				defer func() {
+					tempFile.Close()
+					os.Remove(tempFile.Name())
+				}()
 			}
 
 			files := []cremote.File{
```

/cc @elias-orijtech

--- a/cmd/cosign/cli/policy_init.go
+++ b/cmd/cosign/cli/policy_init.go
@@ -146,7 +146,10 @@ func initPolicy() *cobra.Command {
 					return err
 				}
 				outfile = tempFile.Name()
-				defer os.Remove(tempFile.Name())
+				defer func() {
+					tempFile.Close()
+					os.Remove(tempFile.Name())
+				}()
 			}
 
 			files := []cremote.File{
@@ -309,7 +312,10 @@ func signPolicy() *cobra.Command {
 					return err
 				}
 				outfile = tempFile.Name()
-				defer os.Remove(tempFile.Name())
+				defer func() {
+					tempFile.Close()
+					os.Remove(tempFile.Name())
+				}()
 			}
 
 			files := []cremote.File{

